### PR TITLE
Update/Replace go packages

### LINKED ISSUE #97: Review the golang packages we use
Related to #70 and #34.

The "nothing fancy" or "quick wins" packages:

- [ ] github.com/blang/semver
  - Semver parser.
  - We should update it to github.com/blang/semver/v4, it's the release that's compatible with go modules.

- [x] gopkg.in/fatih/set.v0
  - Just a set library. It's an archived and unmaintained project.
  - Just write our own set difference function.
- [x] github.com/satori/go.uuid
  - Uuid generator and stuff.
  - Using latest release already.
  - Unmaintained and buggy, needs replacing.

The database packages:

- [ ] github.com/jmoiron/sqlx
  - Extended golang's database/sql.
  - Already updated to latest release. We will see if we need it when
    we decide on new ORM.
  - The goqu package also provides some wrappers around database/sql,
    which we don't currently use. We could see if goqu could replace this
    package.

- [x] github.com/lib/pq
  - Postgresql driver.
  - Already at latest release.
  - Replaced with https://github.com/jackc/pgx

- [x] github.com/rubenv/sql-migrate
  - Schema migration thingy.
  - Could be updated, there are no releases, though.

- [x] gopkg.in/mgutz/dat.v1
  - That's an ORM, basically.
  - Alternatives:
    - https://github.com/jinzhu/gorm
    - https://github.com/go-pg/pg
  - We went with https://github.com/doug-martin/goqu

The web framework packages:

- [x] github.com/ymichael/sessions
  - Related to the goji framework, so see below

- [x] github.com/zenazn/goji
  - A web framework.
  - So. Many. Web. Frameworks.
  - https://github.com/mingrammer/go-web-framework-stars

Other packages:

- [x] github.com/aquam8/go-omaha
  - Should use github.com/coreos/go-omaha or our own fork.

- [ ] github.com/facebookgo/grace
  - Used only in the demo.
  - Consider making the demo a separate module maybe.

- [x] github.com/google/go-github
  - Library for talking to github.
  - Needs updating.

- [ ] github.com/mgutz/logxi
  - A logging solution.
  - Use logrus or glog.

- [x] github.com/stretchr/testify
  - Some testing helpers.
  - Already using the latest release.
  - Do we want to use something else?

- [x] golang.org/x/oauth2
  - Client for oauth2.
  - There are no releases, could bump it to latest revision.

### PR DESCRIPTION
**Pull request by ashu8912**
fixes #97

### CODE REVIEW

**Review by krnowak — CHANGES_REQUESTED** [review]
Oh man, logger library change is painful. Thanks for handling it.

So I haven't gone through every single change here, as there are many. But one thing sticks out - our logging is a mess. Both old `logxi` and new `zerolog` are structured loggers, but `logxi` was rather weak at enforcing it, while `zerolog` is more stringent in this regard. So with `logxi` the usual way of passing parameter to logger was `logger.Debug(message, key1, value1, key2, value2, …)`, and we seem to often abuse it in way like `logger.Error("stuff failed", err)`, where err becomes `key1`. With `zerolog`, the way of logging looks more like `logger.Error().Str(key1, value1).Str(key2, value2).Msg(message)`. Or `logger.Error().Err(err).Msg("stuff failed")`. The problem I see here is that in this PR we put all the key-value pairs into the message by turning `logger.Debug("message", key1, value1, key2, value2, …)` into `logger.Debug().Msgf("message %s %s %s %s …", key1, value1, key2, value2, …)`. I'd prefer to keep the key-value and message separation.

So to summarize:
- Please use `Err` for passing errors (which you mostly already do). Not sure if `AnErr` comes in handy anywhere.
- Please use `Str` (or some other type-safe function from https://pkg.go.dev/github.com/rs/zerolog#Event) for key-value pairs.
- Please come up with some guideline for messages (the `Msg` or `Msgf` part). My proposal would be that they are written like "operation - message", so for example "fetching manifest - something bad happened" or "login cb - invalid oauth state".
- The loggers were named (`logger = log.New("auth")`), I suppose this can be useful to preserve (like with `zerolog.New(zerolog.ConsoleWriter{…}).With().Str("context", "auth")`).
- Please fix some of the logging mess/abuse we have.

Some examples:

`logger.Debug("auth metrics", "malformed authorization header", authHeader)` could become `logger.Debug().Str("header", authHeader").Msg("auth metrics - malformed authorization header")`

`logger.Error("login cb", "expected to have a valid desiredurl item in session data")` could become `logger.Error().Msg("login cb - expected to have a valid desiredurl item in session data")`

`logger.Error("addApp - cloning app", "error", err.Error(), "app", app, "sourceAppID", sourceAppID)` could become `logger.Error().Err(err).Str("app", app).Str("sourceAppID", sourceAppID).Msg("addApp - cloning app failed")`

File: update_nebraska_db_ts.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/env bash
```
[the remark this answers is not in the document]
Comment (joaquimrocha) [reply update_nebraska_db_ts.sh:1]: This was probably mistakenly added.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
As Krzesimir had noticed, there's a debug script that has been mistakenly added to the PR.
The choice of Zerolog means changing the logging calls in a lot of places. I am curious to know why it was used over Logrus (it's in no-new-features mode but seems to be actively maintained, and is compatible with the log Go API).

**Review by krnowak — COMMENTED** [review]
I think there is one more thing to do, which is to set the default logging level to "info", and then either add a `debug` flag or some environment variable to set the logging to the "debug" level.

**Review by joaquimrocha — APPROVED** [review]

**Pull request merged**